botocore/handlers.py

```diff
@@ -484,7 +484,7 @@ def parse_get_bucket_location(parsed, http_response, **kwargs):
     # The "parsed" passed in only has the ResponseMetadata
     # filled out.  This handler will fill in the LocationConstraint
     # value.
-    if 'LocationConstraint' in parsed:
+    if 'LocationConstraint' in parsed or 'Error' in parsed:
```

Comment:
What do people think about changing this if statement to 
`if http_response.raw is None:`?

The handler function is only needed to manually parse a raw response into a parsed response since it can't be modeled correctly. The precondition for this function is that this a real response we got from the service. And it has a raw_response, and an incorrect parsed_response that needs to be manually built  from the raw response. It should skip if this was a stubbed response as there is no raw response, only a preset parsed response.

I think it makes more sense to check this as directly as possible. `parse_get_bucket_location` depends on the raw response, so we should just check it. If it isnt there, there is nothing to do. The current if statements have the same effect of skipping the function in the correct cases, but they don't really do a good job of explaining why. They check for the side effects of the case where there is no raw response. We could then update the comment to say that there is no raw response to parse, this is likely a stubbed response.

Reply:
yeah, this makes sense to me. We can directly check for the raw response rather than checking for every side effects of it.